spec/shell_wrapper_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+require 'rspec'
+require 'tmpdir'
+require 'open3'
+
+module SwiftBuildpack
+  module ShellWrapper
+  end
+end
+
+# only run these tests on Linux
+describe SwiftBuildpack::ShellWrapper, :if => (/darwin/ =~ RUBY_PLATFORM).nil? do
+  let(:app_dir) { Dir.mktmpdir }
+  let(:apt_cache_dir) { Dir.mktmpdir }
+  let(:buildpack_dir) { File.join(File.dirname(__FILE__), '../') }
+  let(:common_shell_script) { File.join(buildpack_dir, 'lib', 'common.sh') }
+  let(:default_swift_version) { '3.1' }
+
+  describe '#install_packages' do
+    context 'deb file exists in $APT_CACHE_DIR/archives' do
+      before do
+        ENV['APT_CACHE_DIR'] = apt_cache_dir
+        ENV['BUILD_DIR'] = app_dir
+        system("mkdir -p #{File.join(apt_cache_dir, 'archives')}")
+        cmd = "curl -sSL -o #{File.join(apt_cache_dir, 'archives', 'libpython2.7.deb')} http://security.ubuntu.com/ubuntu/pool/main/p/python2.7/libpython2.7-dev_2.7.6-8ubuntu0.3_amd64.deb"
```

Comment:
The problem is that this spec is testing a bash script which then runs `dpkg -x` on the downloaded file, so a file with an invalid format would fail and we'd need to actually include a real .deb file in the buildpack to make the test pass.  If the download fails, rspec will retry up to 3 times (configured in spec_helper.rb).